graylog2-server/src/main/java/org/graylog/events/indices/MoreIndices.java

```diff
@@ -74,62 +60,26 @@ public void bulkIndex(List<EventWithContext> eventsWithContext) {
-            // to use it for our events as well to make sure we can use the search without errors.
-            source.put(EventDto.FIELD_EVENT_TIMESTAMP, buildElasticSearchTimeFormat(requireNonNull(event.getEventTimestamp()).withZone(UTC)));
-            source.put(EventDto.FIELD_PROCESSING_TIMESTAMP, buildElasticSearchTimeFormat(requireNonNull(event.getProcessingTimestamp()).withZone(UTC)));
-            if (event.getTimerangeStart() != null) {
-                source.put(EventDto.FIELD_TIMERANGE_START, buildElasticSearchTimeFormat(event.getTimerangeStart().withZone(UTC)));
-            }
-            if (event.getTimerangeEnd() != null) {
-                source.put(EventDto.FIELD_TIMERANGE_END, buildElasticSearchTimeFormat(event.getTimerangeEnd().withZone(UTC)));
-            }
-
-            // We cannot index events that don't have any stream set
-            if (event.getStreams().isEmpty()) {
-                throw new IllegalStateException("Event streams cannot be empty");
-            }
-
-            // Collect a set of indices for the event to avoid writing to the same index set twice if
-            // multiple streams use the same index set.
-            final Set<String> indices = event.getStreams().stream()
-                .map(streamId -> {
-                    final String index = streamIndices.get(streamId);
-                    if (index == null) {
-                        LOG.warn("Couldn't find index set of stream <{}> for event <{}> (definition: {}/{})", streamId,
-                            event.getId(), event.getEventDefinitionType(), event.getEventDefinitionId());
-                    }
-                    return index;
+        final List<Map.Entry<String, Event>> requests = eventsWithContext.stream()
+                .flatMap(eventWithContext -> {
+                    final Event event = eventWithContext.event();
+                    // Collect a set of indices for the event to avoid writing to the same index set twice if
+                    // multiple streams use the same index set.
+                    final Set<String> indices = event.getStreams().stream()
+                            .map(streamId -> {
+                                final String index = streamIndices.get(streamId);
+                                if (index == null) {
+                                    LOG.warn("Couldn't find index set of stream <{}> for event <{}> (definition: {}/{})", streamId,
+                                            event.getId(), event.getEventDefinitionType(), event.getEventDefinitionId());
+                                }
+                                return index;
+                            })
+                            .filter(Objects::nonNull)
+                            .collect(Collectors.toSet());
+                    return indices.stream()
+                            .map(index -> new AbstractMap.SimpleEntry<>(index, event));
```

Comment:
I would love it, if you could unpack this by extracting lambda blocks to private methods. Ideally we'd only have one line lambdas, which are much easier to read than these nested blocks.